src/zeroconf/_dns.py

```diff
@@ -487,7 +483,7 @@ def __eq__(self, other: Any) -> bool:
             isinstance(other, DNSNsec)
             and self.next_name == other.next_name
```

Comment:
We can move all the checks after the isinstance into a cdef with the other typed to the same type which will avoid all the python accessor overhead since we can access the struct values in C instead